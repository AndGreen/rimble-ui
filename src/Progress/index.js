import React from 'react'
import styled from 'styled-components'

import { transparentize } from 'polished'

import theme from '../theme'

const Progress = styled.progress`
  & {
    appearance: none;
    display: inline-block;
    overflow: hidden;
    height: 8px;
    border-radius: ${props => props.theme.radii[1]};
  }
  &::-webkit-progress-bar {
    background-color: ${props => transparentize('0.75', props.theme.colors.primary)};
  }
  &::-webkit-progress-value {
    background-color: ${props => props.theme.colors.primary};
    border-radius: ${props => props.theme.radii[1]};
  }
`

Progress.defaultProps = {
  theme: theme
}

Progress.displayName = 'Progress'

export default Progress
