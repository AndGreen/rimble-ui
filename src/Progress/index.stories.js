import React from 'react'

import { storiesOf } from '@storybook/react'

import Progress from '../Progress'

storiesOf('Form/Progress', module)
  .add('default', () => (
    <Progress value={0.5} />
  ))
