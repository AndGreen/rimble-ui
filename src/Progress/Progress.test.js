import Progress from './'

describe('Progress component sanity', () => {
  it('has name', () => {
    expect(Progress.displayName).toBe('Progress')
  })

  it('matches default snapshot', () => {
    const component = render(<Progress />)
    expect(component).toMatchSnapshot()
  })
})
