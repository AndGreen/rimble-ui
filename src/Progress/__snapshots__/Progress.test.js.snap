// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress component sanity matches default snapshot 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  overflow: hidden;
  height: 8px;
  border-radius: 4px;
}

.c0::-webkit-progress-bar {
  background-color: rgba(78,63,206,0.25);
}

.c0::-webkit-progress-value {
  background-color: #4E3FCE;
  border-radius: 4px;
}

<progress
  class="c0"
>
  
</progress>
`;
